src/core/heating_cooling_loads.jl

```diff
@@ -1,249 +1,250 @@
+    demand.
 """
-struct DomesticHotWaterLoad
-    loads_kw::Array{Real, 1}
-    annual_mmbtu::Real
-    unaddressable_annual_fuel_mmbtu::Real
-
-    function DomesticHotWaterLoad(;
-        doe_reference_name::String = "",
-        city::String = "",
-        blended_doe_reference_names::Array{String, 1} = String[],
-        blended_doe_reference_percents::Array{<:Real,1} = Real[],
-        annual_mmbtu::Union{Real, Nothing} = nothing,
-        monthly_mmbtu::Array{<:Real,1} = Real[],
-        addressable_load_fraction::Any = 1.0,
-        fuel_loads_mmbtu_per_hour::Array{<:Real,1} = Real[],
-        time_steps_per_hour::Int = 1, # corresponding to `fuel_loads_mmbtu_per_hour`
-        latitude::Real = 0.0,
-        longitude::Real = 0.0,
-        existing_boiler_efficiency::Real = NaN
+function HeatingLoad(;
+    load_type::String = "",
+    doe_reference_name::String = "",
+    blended_doe_reference_names::Array{String, 1} = String[],
+    blended_doe_reference_percents::Array{<:Real,1} = Real[],
+    industrial_reference_name::String = "",
+    blended_industrial_reference_names::Array{String, 1} = String[],
+    blended_industrial_reference_percents::Array{<:Real,1} = Real[],    
+    city::String = "",
+    year::Int = doe_reference_name ≠ "" || blended_doe_reference_names ≠ String[] ? 2017 : 2022, # CRB profiles must use 2017. If providing load profile, specify year of data.
+    annual_mmbtu::Union{Real, Nothing} = nothing,
+    monthly_mmbtu::Array{<:Real,1} = Real[],
+    addressable_load_fraction::Any = 1.0,
+    fuel_loads_mmbtu_per_hour::Array{<:Real,1} = Real[],
+    normalize_and_scale_load_profile_input::Bool = false,
+    time_steps_per_hour::Int = 1, # corresponding to `fuel_loads_mmbtu_per_hour`
+    latitude::Real = 0.0,
+    longitude::Real = 0.0,
+    existing_boiler_efficiency::Real = NaN
     )
 
-        if length(addressable_load_fraction) > 1
-            if !isempty(fuel_loads_mmbtu_per_hour) && length(addressable_load_fraction) != length(fuel_loads_mmbtu_per_hour)
-                throw(@error("`addressable_load_fraction` must be a scalar or an array of length `fuel_loads_mmbtu_per_hour`"))
-            end
-            if !isempty(monthly_mmbtu) && length(addressable_load_fraction) != 12
-                throw(@error("`addressable_load_fraction` must be a scalar or an array of length 12 if `monthly_mmbtu` is input"))
-            end
-            addressable_load_fraction = convert(Vector{Real}, addressable_load_fraction)
-        elseif typeof(addressable_load_fraction) <: Vector{}
-            addressable_load_fraction = convert(Real, addressable_load_fraction[1])            
-        else
-            addressable_load_fraction = convert(Real, addressable_load_fraction)
+    # Determine which type of heating load to build
+    if load_type == "space_heating"
+        load = :SpaceHeatingLoad
+        struct_type = SpaceHeatingLoad
+    elseif load_type == "domestic_hot_water"
```

Comment:
with this naming convention, should we rename the `dhw_load` object in the scenario object to `domestic_hot_water_load`?  If you agree, I may add that to the other const refactoring in a separate PR so it doesn't block progress on this.  That way, load types would stay consistent

Reply:
I agree with that change. But yeah I just looked at the "search and replace" and it's a lot especially if we want to include updated the names of the results from e.g. "thermal_to_dhw_load..." to "thermal_to_domestic_hot_water_load...". The results would require some renames on the API side as well. For now, let's punt and put on the TODO list.